prusa/link/web/wifi-wizard/src/App.svelte

```diff
@@ -0,0 +1,316 @@
+<script context="module">
+    export const actions = {
+        save: "save",
+        connect: "connect",
+        disconnect: "disconnect",
+        forget: "forget",
+    }
+</script>
+
+<script>
+    import { slide } from 'svelte/transition'
+    import { handleFormData, requestUrl, changeHost, turnOffHotspot, probe } from './utils.js'
+    import { states } from "./StateMachine.svelte"
+    import ConnectForm from './ConnectForm.svelte'
+    import Fetcher from './Fetcher.svelte'
+    import Modal from './Modal.svelte'
+    import StateMachine from './StateMachine.svelte'
+
+    const REDIRECT_DELAY = 5000;
+
+    const hotspotOffStates = [
+        states.ipsReported,
+        states.hotspotHandoff,
+        states.redirectImminent,
+    ]
+
+    let modal;
+    let fetcher;
+    let stateMachine;
+
+    let aps = [];
+    let connectionDetails;
+    let rawConnectionInfo;
+    let probeDetails;
+    let probeResults;
+
+    let state;
+
+    let selectedSSID;
+
+    let availableIps;
+    let redirectIp;
+    let redirectTimeout;
+
+    function selectAp(ap) {
+        selectedSSID = ap.ssid;
+        fetcher.selectAp(ap)
+    }
+
+    function backToWizard() {
+        window.location.href = requestUrl + "/wizard";
+    }
+
+    function connectionChange(e, apToUpdate) {
+        //TODO: update ap
+        let action = e.target.getAttribute("data-action");
+        let newAps = aps;
+        let index = newAps.indexOf(apToUpdate)
+
+        switch (action) {
+            case actions.save:
+            case actions.connect:
+                if (apToUpdate?.state == 0) {
+                    newAps[index].state = 1;
+                }
+                break;
+            case actions.disconnect:
+            case actions.forget:
+                if (apToUpdate?.state == 2) {
+                    newAps[index].state = 3;
+                }
+                break;
+            default:
+                console.log("data-action was not found action: " + action);
+        }
+        aps = newAps;
+        handleFormData(e);
+        setTimeout(fetcher.fetchConnectionInfo, 300);
+    }
+
+    async function autoRedirect(probeDetail) {
+        let reachable = await probe(probeDetail.url);
+        if (reachable) {
+            changeHost(probeDetail.url)
+        } else {
+            redirectIp = undefined;
+        }
+    }
+
+    async function redirectAttempt(receivedProbeResulte) {
+        if (redirectIp !== undefined) {
+            return;
+        }
+        if (state != states.redirectImminent) {
+            return;
+        }
+        let potentialRedirectProbe;
+        for (const probeResult of receivedProbeResulte) {
+            if (probeResult.reachable && !probeResult.sameAsHost) {
+                potentialRedirectProbe = probeResult;
+                break;
+            }
+        }
```

Comment:
`receivedProbeResulte` typo, should be `s` on end

```suggestion
        let potentialRedirectProbe = receivedProbeResulte.find(r => r.reachable && !r.sameAsHost)
```